### PR TITLE
Feat(eos_cli_config_gen): Add schema for interface_groups

### PR DESCRIPTION
**Pull request by mayurgs34**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/<data_model_key>.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/<>.j2` and `templates/documentation/<>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

### Reviewer Checklist

- Reviewer 1:
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2:
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  interface_groups:
+    type: list
+    primary_key: name
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml:9]: This key is a bit special, since it is part of the maintenance feature, so I suggest a special display name:
```suggestion
    display_name: Maintenance Interface Groups
    primary_key: name
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          required: true
+          display_name: Interface-Group name
+        interfaces:
+          type: list
+          items:
+            type: str
+        bgp_maintenance_profiles:
+          type: list
+          items:
+            type: str
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml:27]:
```suggestion
            type: str
            description: Name of BGP Maintenance Profile
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        interface_maintenance_profiles:
+          type: list
+          items:
+            type: str
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml:32]:
```suggestion
            type: str
            description: Name of Interface Maintenance Profile
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      type: dict
+      keys:
+        name:
+          type: str
+          required: true
+          display_name: Interface-Group name
+        interfaces:
+          type: list
+          items:
+            type: str
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/interface_groups.schema.yml:22]:
```suggestion
            type: str
            description: Interface Name
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml [hunk 5]
```diff
@@ -93,22 +93,47 @@ keys:
...
-    description: "AVD supports 2 different data models for community lists:\n\n- The
-      legacy `community_lists` data model that can be used for compatibility with
-      the existing deployments.\n- The improved `ip_community_lists` data model.\n\nBoth
-      data models can coexist without conflicts, as different keys are used: `community_lists`
-      vs `ip_community_lists`.\nCommunity list names must be unique.\n\nThe improved
-      data model has a better design documented below:\n\ncommunities and regexp MUST
-      not be configured together in the same entry\npossible community strings are
-      (case insensitive):\n - GSHUT\n - internet\n - local-as\n - no-advertise\n -
-      no-export\n - <1-4294967040>\n - aa:nn\n"
+    description: "AVD supports 2 different data models for community lists:\n\n- The\
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: I am not sure why these changes show up in this PR. Maybe it is something with versions? Please check `ansible --version` if you have `libyaml` installed.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml (hunk 5 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: I approved the PR, but please still look into this. We might have to implement some changes to the schema generation to avoid such changes

Comment (mayurgs34) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: yes sure, I am looking into the issue.

Comment (mayurgs34) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: Hi, This issue is resolved it required python version 3.8 and I was using 3.10. and also 
it requires ansible-core 2.12.6 to get the same format.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: I think it is the ansible version more than the python version making this difference. I run py3.10 as well.

**Review by carlbuchmann — APPROVED** [review]
LGTM!
